src/MosaicViews.jl

```diff
@@ -260,41 +289,41 @@ function mosaicview(A::AbstractArray{T,3},
     MosaicView{T,4}(A_new, dims)
 end
 
-function mosaicview(A::AbstractArray{T,N},
-                    args...;
-                    npad = 0,
+function mosaicview(A::AbstractArray{T,N};
                     nrow = -1,
                     ncol = -1,
-                    rowmajor = false) where {T,N}
+                    kwargs...) where {T,N}
     3 <= N || throw(ArgumentError("The given array must have dimensionality of N=3 or higher"))
     # if neither nrow nor ncol is provided then automatically choose
     # nrow and ncol to reflect what MosaicView does (i.e. use size)
     if nrow == -1 && ncol == -1
         nrow = size(A, 3)
         # ncol = size(A, 4)
     end
-    mosaicview(reshape(A, (size(A,1), size(A,2), :)), args...;
-               npad=npad, nrow=nrow, ncol=ncol, rowmajor=rowmajor)
+    mosaicview(reshape(A, (size(A,1), size(A,2), :));
+               nrow=nrow, ncol=ncol, kwargs...)
 end
 
-function mosaicview(As::AbstractVector{T},
-                    fill=zero(eltype(first(As)));
+mosaicview(As::AbstractArray...; kwargs...) = mosaicview(As; kwargs...)
+
+function mosaicview(As::AbstractVector{T};
+                    fillvalue=zero(eltype(first(As))),
                     kwargs...) where {T <: AbstractArray}
     length(As) == 0 && throw(ArgumentError("The given vector should not be empty"))
     N = ndims(first(As))
     2 <= N || throw(ArgumentError("The given array must have dimensionality of N=2 or higher"))
-    mosaicview(_padded_cat(As; fill=fill, dims=N+1), fill; kwargs...)
+    mosaicview(_padded_cat(As; fillvalue=fillvalue, dims=N+1); fillvalue=fillvalue, kwargs...)
 end
 
-function mosaicview(As::Tuple{T, Vararg{T}},
```

Comment:
The `Tuple{T, Vararg{T}}` annotation becomes too strict that each image should have the same type, but generally, this function supports mixed Gray and RGB images.